services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/Incident.java

```diff
@@ -0,0 +1,416 @@
+   * @return the type adapter for this class
+   */
+  public static TypeAdapter<Incident> typeAdapter(Gson gson) {
+    return new AutoValue_Incident.GsonTypeAdapter(gson);
+  }
+
+  /**
+   * Create a new instance of this class by passing in a formatted valid JSON String.
+   *
+   * @param json a formatted valid JSON string defining an Incident
+   * @return a new instance of this class defined by the values passed in the method
+   */
+  public static Incident fromJson(String json) {
+    GsonBuilder gson = new GsonBuilder();
+    gson.registerTypeAdapterFactory(DirectionsAdapterFactory.create());
+    return gson.create().fromJson(json, Incident.class);
+  }
+
+  /**
+   * This builder can be used to set the values describing the {@link Incident}.
+   */
+  @AutoValue.Builder
+  public abstract static class Builder {
+
+    /**
+     * Unique identifier for incident. It might be the only one <b>non-null</b> filed which meant
+     * that incident started on previous leg and one has an incident with the same <b>id</b>.
+     *
+     * @param id String
+     */
+    public abstract Builder id(@NonNull String id);
+
+    /**
+     * One of incident types.
+     *
+     * @param type incident type
+     * @see IncidentType
+     */
+    public abstract Builder type(@Nullable @IncidentType String type);
+
+    /**
+     * <b>True</b> if road is closed and no possibility to pass through there. <b>False</b>
+     * otherwise.
+     *
+     * @param closed is way closed
+     */
+    public abstract Builder closed(@Nullable Boolean closed);
+
+    /**
+     * Quantitative descriptor of congestion.
+     *
+     * @param congestion congestion
+     */
+    public abstract Builder congestion(@Nullable Congestion congestion);
+
+    /**
+     * Human-readable description of the incident suitable for displaying to the users.
+     *
+     * @param description incident description
+     */
+    public abstract Builder description(@Nullable String description);
+
+    /**
+     * Human-readable long description of the incident suitable for displaying to the users.
+     *
+     * @param longDescription incident long description
+     */
+    public abstract Builder longDescription(@Nullable String longDescription);
+
+    /**
+     * Severity level of incident.
+     *
+     * @param impact impact type
+     * @see ImpactType
+     */
+    public abstract Builder impact(@Nullable @ImpactType String impact);
+
+    /**
+     * Sub-type of the incident.
+     *
+     * @param subType syp-type
+     */
+    public abstract Builder subType(@Nullable String subType);
+
+    /**
+     * Sub-type-specific description.
+     *
+     * @param subTypeDescription sub-type description
+     */
+    public abstract Builder subTypeDescription(@Nullable String subTypeDescription);
+
+    /**
+     * AlertC codes.
+     *
+     * @param alertcCodes list of alert codes
+     * @see <a href="https://www.iso.org/standard/59231.html">AlertC</a>
```

Comment:
NIT Should we use `{@link}` instead?

Reply:
a http(s) link is not working with {@link}